### ISSUE TITLE
[bug] Terminal SIGINT not handled gracefully during .input() calls

### ISSUE THREAD

**Issue by christophercolumbusdog** [fadeevab/cliclack#84]
I noticed in my Rust CLI project that there's an undesirable behavior when using `.input()` and then sending a SIGINT via CTRL+C on MacOS.

Take the following Rust code:
```rust
use cliclack::input;

fn main() {
    let result: String = input("Enter Stuff:")
        .interact()
        .unwrap();
    println!("You entered {}", result);
}
```

Using the following Cargo.toml:
```toml
[package]
name = "rust"
version = "0.1.0"
edition = "2021"

[dependencies]
cliclack = "0.3.5"
```

If, while on running the application and entering text into the prompt, you press `CTRL+C` on the built-in Mac Terminal, the terminal cursor disappears, becoming "stuck" in the expected input:

<img width="399" alt="image" src="https://github.com/user-attachments/assets/b12f85b8-9936-4e9e-971f-8b7c0a5c60be" />

You can't see it in the screenshot, but there is no blinking cursor on any of the subsequent lines for terminal input, requiring you to open a new tab to reset the cursor.

It's not clear to me how much this is due to Cliclack or Terminal, but regardless, I think this indicates that Cliclack needs better support for handling SIGINT during input.

**Comment by fadeevab** [comment]
Yes, it's a known problem across all console and terminal libs (e.g. `dialoguer`, `ratatui`). Use the following [trick](https://github.com/fadeevab/cliclack/blob/main/examples/basic.rs#L13) to handle it. Another similar example: https://github.com/fadeevab/cliclack/blob/main/examples/spinner.rs.

**Comment by christophercolumbusdog** [comment]
Ah, I had not seen that in the examples. Apologies for that; thanks for the response! This addresses my concerns.

**Comment by fadeevab** [comment]
@christophercolumbusdog No worries; let me know if you have any issues.  Basically, many real-world use cases and solutions can be found in the examples.

**Comment by fadeevab** [comment]
Fixed Ctrl-C handling in `cliclack v0.5.5` (PR #112) so the previous workaround is no longer needed.
